### ISSUE TITLE
`ResourceLoader.load_threaded_request` causes Vulkan errors

### ISSUE THREAD

**Issue by lyuma** [godotengine/godot#56776]
### Godot version

4.0.dev 49482966ec0e5b9a8937eef5ef664c9a350f50d3

### System information

Windows 10, Vulkan, NVidia RTX 2080 Ti, Driver 496.13

### Issue description

Loading a PackedScene resource from a thread often appears to trigger errors in the Vulkan renderer and causes random meshes to be missing.

Here are some errors. Ignore "Rect2 size is negative,", "Condition "p_from_line > p_to_line" is true" and "Trying to remove non-existent autoload" since these are red herrings. The errors to look for are Vulkan related.

```
ERROR: Creating buffers with data is forbidden during creation of a draw list
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4146)
ERROR: Condition "!index_buffer_owner.owns(p_index_buffer)" is true. Returning: RID()
   at: RenderingDeviceVulkan::index_array_create (drivers\vulkan\rendering_device_vulkan.cpp:4199)
ERROR: Buffer is either invalid or this type of buffer can't be retrieved. Only Index and Vertex buffers allow retrieving.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6086)
ERROR: Size of vertex data provided (0) does not match expected (247300)
   at: (servers\rendering\renderer_rd\renderer_storage_rd.cpp:3158)
```

The Vulkan errors occur during the load process, before the nodes are even added to the scene. It is random, and sometimes no errors occur, but in the attached project, it seems to fail roughly 80% of the time on my computer. We tried different caching options with no effect. The bug does not occur when using the editor GUI to load the scn, nor at runtime if loaded from the main thread (e.g. with the scene play button).

We've been able to reproduce this with multiple people on several godot master versions since summer, on multiple hardware configurations
![Picture of Sponza with gap in wall](https://media.discordapp.net/attachments/659137056991346708/931358926363045928/unknown.png)

### Steps to reproduce

1. Open the attached project.
2. Play the project.
3. Check the Debugger tab or the windows terminal, as errors do not show in the Output tab for some reason. You may also see some meshes missing between test runs.

### Minimal reproduction project

https://cdn.discordapp.com/attachments/659137056991346708/931360798037655642/backgroundloadbug.zip

**Comment by brianwinterpixel** [comment]
Likely fixed by #54890

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/54151.
